### PR TITLE
Fix Build Warnings (part 6): Add XML comments and remove unused variables in Csharp code

### PR DESCRIPTION
**Pull request by hebinhuang**
Fix Build Warnings (part 6): Add XML comments and remove unused variables in the files below
        Partitioner.cs, StatCounter.cs, StatusTracker.cs, JvmObjectReference.cs, SerDe.cs, DataFrame.cs, Functions.cs, MapWithStateDStream.cs, DStreamStateSample.cs, MessagePublisher.cs

### CODE REVIEW

File: examples/Streaming/Kafka/MessagePublisher.cs [hunk 1]
```diff
@@ -37,7 +37,6 @@ private static void SendMessageBatch(object state)
             });
             var kafkaProducer = new Producer(kafkaProducerConfig);
 */
-            var topicName = "<kafkaTopicName>";
```
Comment (skaarthik) [inline examples/Streaming/Kafka/MessagePublisher.cs]: This is used in line #47 that is commented out because kafkanet library is not yet available in NuGet. Please revert this change

Comment (hebinhuang) [inline examples/Streaming/Kafka/MessagePublisher.cs]: Reverted

**Comment by hebinhuang** [comment]
Thanks Kaarthik for the merge:)
